src/main/java/io/pravega/connectors/flink/serialization/PravegaDeserializationSchema.java

```diff
@@ -126,6 +130,18 @@ public boolean isEndOfStream(T nextElement) {
         return serializer;
     }
 
+    /**
+     * An method for applying the metadata in deserialization.
+     * Override it in the custom extended {@link PravegaDeserializationSchema} if the Pravega metadata is needed.
+     *
+     * @param eventRead The EventRead structure the client returns which contains metadata
+     *
+     * @return the deserialized event with metadata
+     */
+    public T deserializeWithMetadata(EventRead<T> eventRead) {
```

Comment:
This name suggests that it replaces `deserialize` which is not the case. This method does not actually deserialize. I recommending renaming it to `insertMetadata` or something similar.

Reply:
It's true that this one does not do `deserialization`, but it really "get" the data. It has a default implementation `eventRead.getEvent()` and gets called in the `FlinkPravegaReader`  here: https://github.com/pravega/flink-connectors/pull/305/files#diff-005207bb16199a05351084fea79fd9acR271

Reply:
I'm also not a fan of the current name, but I don't think `insertMetadata` is correct because if it's not overriden which is most cases, it won't insert metadata. Maybe `getData` , `getEvent` is better? Any suggestions of a better name?

Reply:
When reviewing this code I also got confused by the name of this method because it's not actually obvious that both `deserialize` methods will get called, one at the Pravega level and one at the Flink Connector level.  

It's almost like the method should be called `extractEvent` since it's extracting it from the Pravega level API (and providing the chance to inject Metadata into the event).  I considered perhaps it should be called `getEvent` or `retrieveEvent` but this would get confusing with the Pravega semantics.

Reply:
`extractEvent` is a good one, I can buy in this name